connectors/db/vector.py

```diff
@@ -284,6 +402,93 @@ def add_file(self, file: File, agent_id: int):
             except Exception:
                 log.exception("error adding documents to vector store for batch %d", current_batch)
 
+    def deduplicate_results(self, results, threshold=0.90):
+        """
+        Removes near-duplicate search results based on text similarity.
+        
+        - `threshold=0.90` means chunks with >= 90% text similarity are considered duplicates.
+        - Keeps only the highest-ranked unique chunk.
+        """
+        if not results:
+            return []
+
+        vectorizer = TfidfVectorizer().fit_transform([r.document.page_content for r in results])
+        similarities = cosine_similarity(vectorizer)
+
+        unique_results = []
+        seen_indices = set()
```

Comment:
Trying to think of a name that better describes what this is tracking... what about `flagged_indices` or `duplicate_result_indices`?

Reply:
I pushed up `indicies_of_dups`? That seem OK?